core/commands/name/name.go

```diff
@@ -62,5 +77,159 @@ Resolve the value of a dnslink:
 		"publish": PublishCmd,
 		"resolve": IpnsCmd,
 		"pubsub":  IpnsPubsubCmd,
+		"inspect": IpnsInspectCmd,
+	},
+}
+
+type IpnsInspectValidation struct {
+	Valid  bool    `json:"valid"`
+	Reason string  `json:"reason,omitempty"`
+	Key    peer.ID `json:"key"`
+}
+
+type IpnsInspectEntry struct {
+	Value        string                          `json:"value"`
+	ValidityType *ipns_pb.IpnsEntry_ValidityType `json:"validityType"`
+	Validity     *time.Time                      `json:"validity"`
+	Sequence     uint64                          `json:"sequence"`
+	TTL          *uint64                         `json:"ttl"`
+	PublicKey    string                          `json:"publicKey"`
+	Signature    string                          `json:"signature"`
+	Data         interface{}                     `json:"data"`
+}
+
+type IpnsInspectResult struct {
+	Entry      IpnsInspectEntry       `json:"entry"`
+	Validation *IpnsInspectValidation `json:"validation,omitempty"`
+}
+
+var IpnsInspectCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Inspects an IPNS Record",
+	},
+	Arguments: []cmds.Argument{
+		cmds.FileArg("record", true, false, "The path to a file with IPNS record to be verified.").EnableStdin(),
+	},
+	Options: []cmds.Option{
+		cmds.StringOption("verify", "The IPNS key to validate against."),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		file, err := cmdenv.GetFileArg(req.Files.Entries())
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		var b bytes.Buffer
+
+		_, err = io.Copy(&b, file)
+		if err != nil {
+			return err
+		}
+
+		var entry ipns_pb.IpnsEntry
+		err = proto.Unmarshal(b.Bytes(), &entry)
+		if err != nil {
+			return err
+		}
+
+		result := &IpnsInspectResult{
+			Entry: IpnsInspectEntry{
+				Value:        string(entry.Value),
+				ValidityType: entry.ValidityType,
+				Sequence:     *entry.Sequence,
+				TTL:          entry.Ttl,
+				PublicKey:    string(entry.PubKey),
+				Signature:    string(entry.SignatureV2),
+				Data:         nil,
+			},
+		}
+
+		if len(entry.SignatureV2) == 0 {
+			result.Entry.Signature = string(entry.SignatureV1)
+		}
```

Comment:
:warning:  Let's not be too smart here. The goal of this command is inspection of record, so no obfuscation.
Let's use exactly the same names of fields as in:
https://github.com/ipfs/specs/blob/main/ipns/IPNS.md#record-serialization-format
and have separate `signatureV1` and `signatureV2`, so it is possible to tell when V1 is missing :pray: